Source/Development_5_upgrade.py

```diff
@@ -0,0 +1,160 @@
+      card = []
+      level = 0
+      for line in cards.readlines():
+        if len(line) == 1:
+          cardImages.append(card)
+          level = 0
+          card = []
+          continue
+        card.append(line.replace("\n",""))
+        level += 1
+      cardImages.append(card)
+    
+    deck = []
+    index = 0
+    for suit in suits:
+      for value in values:
+        deck.append(Card(suit, value, cardImages[index], cardBack))
+        index += 1
+    
+    self.cards = deck
+    self.size = len(deck)
+    self.cardBack = cardBack
+    self.discarded = []
+
+  def shuffle(self):
+    random.shuffle(self.cards)
+
+  def getCard(self):
+    card = self.cards.pop()
+    self.size -= 1
+    self.discarded.append(card)
+    return card
+
+class Player:
+  def __init__(self, name, money: int = 0):
+    self.name = name
+    self.hand = []
+    self.knownCards = []
+    self.money = money
+
+  def addMoney(self, amount: int):
+    self.money += amount
+    return self.money
+
+  def makeBet(self, amount: int):
+    if amount > self.money:
+      print("%s does not have enough money to make this bet." % self.name)
+      return self.money
+    self.money -= amount
+    return self.money
+
+  def addCard(self, card: Card, isKnown: bool = True):
+    self.hand.append(card)
+    if isKnown:
+      self.knownCards.append(True)
+    else:
+      self.knownCards.append(False)
+
+  def clearHand(self):
+    self.hand = []
+    self.knownCards = []
+
+PlayerList = List[Player]
+
+class Dealer:
+  def __init__(self):
+    pass
+
+  def printCards(self, cards: CardList, showFront: bool, printShort: bool = True):
+    for idx in range(6):
+      for i, card in enumerate(cards):
+        if printShort and i < len(cards)-1:
+          image = card.shortImage[idx] if showFront else card.cardBack[idx]
+          print(image, end="")
+        else:
+          image = card.image[idx] if showFront else card.cardBack[idx]
+          print(image, end="")
+      print()
+    
+  def printPlayerCards(self, player: Player, printShort: bool = False):
+    for idx in range(6):
+      for i, card in enumerate(player.hand):
+        if printShort and i < len(player.hand)-1:
+          image = card.shortImage[idx]  if player.knownCards[i] else card.cardBack[idx]
+          print(image, end="")
+        else:
+          image = card.image[idx] if player.knownCards[i] else card.cardBack[idx]
+          print(image, end="")
+      print()
+
+  def dealCards(self, numCards: int, players: PlayerList, deck: Deck):
+    if numCards * len(players) > deck.size:
+      return False
+    for player in players:
+      for _ in range(numCards):
+        player.addCard(deck.getCard())
+    return True
+#Works Cited: StackOverflow
+
+
+class PlayBluff:
+  def __init__(self, bluff, discard, card_list):
+    bluff = True
+    discard = []
+    card_list = []
+
+  def our_decks(discard, bluff, card_list):
+    main_deck = Deck()
+    if bluff:
+      discard == card_list
```

Comment:
This is not proper syntax as this is a conditional statement, not assignment.

Reply:
Thanks for letting me know Prof! I'll see if I can adjust it to fit the assignment better